src/main/java/duke/command/Statement.java

```diff
@@ -0,0 +1,145 @@
+package duke.command;
+
+import duke.exception.DukeException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.Scanner;
+
+/**
+ * Statement class deals with the user's input and make sense of it.
+ * Deals with most of the input errors. Has a table that store number of arguments needed
+ * for each command.
+ */
+public class Statement {
+    private final String statement;
+    private final Hashtable<String, Integer> argsTable;
+
+    /**
+     * Returns a Statement object that represent the user's input.
+     *
+     * @param statement The user's current input.
+     */
+    public Statement(String statement) {
+        this.statement = statement;
+        argsTable = new Hashtable<>();
+
+        argsTable.put("list", 0);
+        argsTable.put("bye", 0);
+        argsTable.put("done", 1);
+        argsTable.put("todo", 1);
+        argsTable.put("find", 1);
+        argsTable.put("delete", 1);
+        argsTable.put("deadline", 2);
+        argsTable.put("event", 2);
+    }
+
+    //return a list of String with the first element as the command
+    //the second element is description and third element is time (if applicable)
+    private List<String> parseStatement() throws DukeException {
```

Comment:
The method here is a bit too long. Would it be possible to make this method slightly shorter by adding helper methods?